### ISSUE TITLE
no_std builds are broken on crates.io stable2407

### ISSUE THREAD

**Issue by seunlanlege** [paritytech/polkadot-sdk#5350]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Description of bug

Updating a runtime to the latest release on crates.io for v1.15.0 results in the following error

```
  error[E0152]: duplicate lang item in crate `std` (which `memchr` depends on): `panic_impl`.
    |
    = note: the lang item is first defined in crate `sp_io` (which `frame_support` depends on)
    = note: first definition in `sp_io` loaded from /home/seun/Projects/hyperbridge/target/wasm32-unknown-unknown/release/wbuild/gargantua-runtime/target/wasm32-unknown-unknown/release/deps/libsp_io-738828916d12a9a6.rmeta
    = note: second definition in `std` loaded from /home/seun/Projects/hyperbridge/target/wasm32-unknown-unknown/release/wbuild/gargantua-runtime/target/wasm32-unknown-unknown/release/deps/libstd-4757ea32571197a0.rmeta

  For more information about this error, try `rustc --explain E0152`.
  error: could not compile `frame-support` (lib) due to 1 previous error
  warning: build failed, waiting for other jobs to finish...
warning: build failed, waiting for other jobs to finish...

```

### Steps to reproduce

Try to build the parachain runtime in this repo: https://github.com/polytope-labs/sc-simnode/

You will run into this exact error

https://github.com/polytope-labs/sc-simnode/actions/runs/10319394800/job/28567692044

**Comment by seunlanlege** [comment]
Digging a bit deeper the problem seems to come from docify??

cargo tree shows 


```
├── frame-support v37.0.0
│   │   │   ├── aquamarine v0.5.0 (proc-macro) (*)
│   │   │   ├── array-bytes v6.2.2
│   │   │   ├── bitflags v1.3.2
│   │   │   ├── docify v0.2.8
│   │   │   │   └── docify_macros v0.2.8 (proc-macro)
│   │   │   │       ├── common-path v1.0.0
│   │   │   │       ├── derive-syn-parse v0.2.0 (proc-macro)
│   │   │   │       │   ├── proc-macro2 v1.0.81 (*)
│   │   │   │       │   ├── quote v1.0.36 (*)
│   │   │   │       │   └── syn v2.0.60 (*)
│   │   │   │       ├── once_cell v1.19.0
│   │   │   │       ├── proc-macro2 v1.0.81 (*)
│   │   │   │       ├── quote v1.0.36 (*)
│   │   │   │       ├── regex v1.10.4 <- regex ??
│   │   │   │       │   ├── aho-corasick v1.1.3
│   │   │   │       │   │   └── memchr v2.7.4
│   │   │   │       │   ├── memchr v2.7.4
```

Docify depends on regex with `default-features = true`

https://github.com/sam0x17/docify/blob/main/macros/Cargo.toml

and regex pulls in std

https://github.com/rust-lang/regex/blob/aa2d8bd/Cargo.toml

**Comment by bkchr** [comment]
Downgrade `serde_json` to fix it for now. I fixed it properly here: https://github.com/paritytech/polkadot-sdk/commit/4def82e7ff6cfacee9e33f53e20f25c10ce6b9e9

**Comment by seunlanlege** [comment]
> Downgrade `serde_json` to fix it for now. I fixed it properly here: [4def82e](https://github.com/paritytech/polkadot-sdk/commit/4def82e7ff6cfacee9e33f53e20f25c10ce6b9e9)

You were right, `serde_json` was the culprit. This command fixed my issue for anyone else who happens on this thread

```
cargo update serde_json --precise 1.0.120
```
